src/components/Form/Form.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+
+interface LoginFormProps {
+  onSubmit?: (username: string, password: string) => Promise<void>;
```

Comment:
The submit function should be managed by the component itself, there is no reason for receiving it as a prop.